packages/connect-webextension/src/popup.ts

```diff
@@ -38,7 +52,13 @@ export class PopupManager extends EventEmitter {
         this.channel.init();
     }
 
-    request() {
+    async request() {
+        const tabExists = await checkIfTabExists(this.popupWindow?.id);
+        if (this.popupWindow?.id && !tabExists) {
+            // We have an id to a non-existing tab, so we unlock it before continue.
+            this.unlock();
+        }
```

Comment:
This is me just experimenting trying to unlock popup if there is no tab, since the first call works as expected but after the first call it fails. It looks like popup is not properly cleared. I am still investigating how to do it properly since it does not work yet. @mroz22 maybe you have some idea.